### ISSUE TITLE
BaseActiveFixture have $_models set as private, making it hard to overwrite.

### ISSUE THREAD

**Issue by agostlg** [yiisoft/yii2#10050]
I need to overwrite getModel($name) but I can't because it make use of $_models. The only way I could think of is to overwride also all other methods that use $_methods so I can create my own $_model on child class with protected access.

**Comment by samdark** [comment]
What's your use case for overiding it?

**Comment by agostlg** [comment]
The fixture uses a different model that have findOne($condition) as a non-static method, and it's important for me to overwrite getModel($name)  to use $model->findOne($keys) instead of $modelClass::findOne($keys). 

My model does not extend any class, it's a MultipleActiveRecord that I created to handle reading from one source and writing on another (e.g.1 reads from ES and updates mysql with the possibility of updating ES also on runtime or sending an message to the broker for later process).  To make implementation simpler and to be able to test it well I created findOne and findAll methods as non-static and they will redirect to the reader (ES in the example 1) and I changed the used static calls on my project to not use it static anymore, but getModel($name) is part of Yii code so I need to override in order to change that.

So I fixed that here by overwriting also other methods, but if $_models was protected it would be easier.
